framework/src/meshmodifiers/MeshModifier.C

```diff
@@ -19,6 +19,9 @@ InputParameters validParams<MeshModifier>()
 {
   InputParameters params = validParams<MooseObject>();
   params.addParam<std::vector<std::string> >("depends_on", "The MeshModifiers that this modifier relies upon (i.e. must execute before this one)");
+  params.addParam<bool>("force_prepare", false,
```

Comment:
Does this need to be user-settable?  When will the user know to turn it on or off?

I think this should be a flag set in the constructor of the class for each MeshModifier.